### PR TITLE
[Bug][Cache] Add data_size field for InternalService.PCacheValue

### LINKED ISSUE #6044: [Bug] Partition Cache rpc miss data_size field
**Describe the bug**

When partition cache is enabled, a query hitting partition cache reported error.

`ERROR 1064 (HY000): Message missing required fields: data_size`

**To Reproduce**
1. create a table:

CREATE TABLE `table2` (
  `event_day` date NULL COMMENT "",
  `siteid` int(11) NULL DEFAULT "10" COMMENT "",
  `citycode` smallint(6) NULL COMMENT "",
  `username` varchar(32) NULL DEFAULT "" COMMENT "",
  `pv` bigint(20) SUM NULL DEFAULT "0" COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`event_day`, `siteid`, `citycode`, `username`)
COMMENT "OLAP"
PARTITION BY RANGE(`event_day`)
(PARTITION p201706 VALUES [('0000-01-01'), ('2017-07-01')),
PARTITION p201707 VALUES [('2017-07-01'), ('2017-08-01')),
PARTITION p201708 VALUES [('2017-08-01'), ('2017-09-01')))
DISTRIBUTED BY HASH(`siteid`) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "V2"
); 

2.  insert some data

`insert into table2 values('2017-08-03', 100, 200, 'test', 400);`

3.  enable partition cache

`set enable_partition_cache=true;`

4.  execute a query

select event_day from table2 where event_day > '2017-06-03' and event_day <= '2017-08-03' group by event_day;

5. see the error

`ERROR 1064 (HY000): Message missing required fields: data_size`

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

`data_size` is a required field for `InternalService.PCacheValue`.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #6044) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/cache/RowBatchBuilder.java [hunk 1]
```diff
@@ -167,13 +167,19 @@ public void copyRowData(RowBatch rowBatch) {
...
                     .addValues(InternalService.PCacheValue.newBuilder()
                             .setParam(InternalService.PCacheParam.newBuilder()
                                     .setPartitionKey(key)
                                     .setLastVersion(partition.getPartition().getVisibleVersion())
                                     .setLastVersionTime(partition.getPartition().getVisibleVersionTime())
-                                    .build()).addAllRows(
+                                    .build())
+                            .setDataSize(dataSize)
+                            .addAllRows(
                                     partitionRowList.stream().map(row -> ByteString.copyFrom(row))
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/qe/cache/RowBatchBuilder.java:183]: The indentation is strange

**Comment by EmmyMiao87** [comment]
Please add a issue about before this PR what happen will be caused.

**Review by morningman — APPROVED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**